docs/src/types.md

```diff
@@ -0,0 +1,162 @@
+```@meta
+CurrentModule = GeometricAlgebra
+```
+
+# Types
+
+## Type Hierarchy
+
+The relationships between GeometricAlgebra types mirror the hierarchical relationships
+between the mathematical objects they represent.
+
+* `AbstractMultivector` represents an arbitrary multivector. All concrete GeometricAlgebra
+  types are subtypes of `AbstractMultivector`.
+
+* `AbstractBlade` represents an arbitrary blade. Any mathematical objects that is logically
+  a blade (i.e., a blade, a pseudoscalar, or a scalar) is represented by a concrete
+  GeometricAlgebra type that is a subtype of `AbstractBlade`.
+
+* `AbstractScalar` represents an arbitrary scalar value. Any mathematical objects that is
+  logically scalar (i.e., a scalar or a special scalar value) is represented by a concrete
+  GeometricAlgebra type that is a subtype of `AbstractScalar`.
+
+```
+AbstractMultivector
+├─ Multivector
```

Comment:
While `Multivector` is in the package, we don't fully support it yet, so do we want to include it in the docs yet?

Reply:
Added warning to let people know that the `Multivector` type is not yet fully implemented.